common/djangoapps/auth_exchange/tests/test_views.py

```diff
@@ -113,3 +118,38 @@ class AccessTokenExchangeViewTestGoogle(
     Tests for AccessTokenExchangeView used with Google
     """
     pass
+
+
+@unittest.skipUnless(settings.FEATURES.get("ENABLE_OAUTH2_PROVIDER"), "OAuth2 not enabled")
+class TestLoginWithAccessTokenView(TestCase):
+    """
+    Tests for LoginWithAccessTokenView
+    """
+    def setUp(self):
+        super(TestLoginWithAccessTokenView, self).setUp()
+        self.user = UserFactory()
+        self.oauth2_client = Client.objects.create(client_type=provider.constants.CONFIDENTIAL)
+
+    def _verify_response(self, access_token, expected_status_code, expected_num_cookies):
+        """
+        Calls the login_with_access_token endpoint and verifies the response given the expected values.
+        """
+        url = reverse("login_with_access_token")
+        response = self.client.post(url, HTTP_AUTHORIZATION="Bearer {0}".format(access_token))
+        self.assertEqual(response.status_code, expected_status_code)
+        self.assertEqual(len(response.cookies), expected_num_cookies)
+
+    def test_success(self):
+        access_token = AccessToken.objects.create(
+            token="test_access_token",
+            client=self.oauth2_client,
+            user=self.user,
+        )
+        self._verify_response(access_token, expected_status_code=204, expected_num_cookies=1)
+        self.assertEqual(len(self.client.cookies), 1)
+        self.assertIsNotNone(self.client.session.session_key)
```

Comment:
Didn't work exactly.  But I updated it to verify the user ID, although it may break the abstraction of the session object, since I'm accessing with a "hidden" key.